src/Grpc.AspNetCore.Server/Internal/GrpcEventSource.cs

```diff
@@ -136,31 +136,31 @@ protected override void OnEventCommand(EventCommandEventArgs command)
                 // This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
                 // They aren't disabled afterwards...
 
-                _totalCallsCounter ??= new PollingCounter("total-calls", this, () => _totalCalls)
+                _totalCallsCounter ??= new PollingCounter("total-calls", this, () => Volatile.Read(ref _totalCalls))
```

Comment:
It's about `Interlocked.Read` vs. `Volatile.Read`.
In ASP.NET Core for the kestrel counters I've used `Volatile.Read`, which does the correct read, then opened https://github.com/dotnet/docs/pull/20984 to update the docs. The consensus over there is to use `Interlocked.Read` as it's documented to prevent torn-reads.

I'm fine with either variant here, just wanted to point this fact out (and for completeness).